`read_excel` errors when `sheet_name` is `None`

**Describe the bug**
Call `read_excel` with `sheet_name` set to `None`. Pyright and Mypy report a type error for the parameter.

Pyright:

```
error: Argument of type "None" cannot be assigned to parameter "sheet_name" of type "int | str" in function "read_excel"
    Type "None" cannot be assigned to type "int | str"
      Type "None" cannot be assigned to type "int"
      Type "None" cannot be assigned to type "str" (reportGeneralTypeIssues)
```

Mypy (from the panda-stubs `poe test_all`)

```
===========================================
Beginning: 'Run Mypy Against Source Code'
===========================================

tests/test_frame.py:1026: error: No overload variant of "read_excel" matches argument types "str", "None"  [call-overload]
tests/test_frame.py:1026: note: Possible overload variants:
tests/test_frame.py:1026: note:     def read_excel(filepath: str, sheet_name: List[Union[int, str]], header: Union[int, Sequence[int], None] = ..., names: Optional[List[str]] = ..., index_col: Union[int, Sequence[int], None] = ..., usecols: Union[int, str, Sequence[Union[int, str, Callable[..., Any]]], None] = ..., squeeze: bool = ..., dtype: Union[str, Dict[str, Any], Union[ExtensionDtype, Union[str, dtype[generic], Type[str], Type[float], Type[int], Type[complex], Type[bool], Type[object]]]] = ..., engine: Optional[str] = ..., converters: Optional[Dict[Union[int, str], Callable[..., Any]]] = ..., true_values: Optional[Sequence[Union[str, bytes, date, datetime, timedelta, bool, int, float, complex, Timestamp, Timedelta]]] = ..., false_values: Optional[Sequence[Union[str, bytes, date, datetime, timedelta, bool, int, float, complex, Timestamp, Timedelta]]] = ..., skiprows: Union[Sequence[int], int, Callable[..., Any], None] = ..., nrows: Optional[int] = ..., na_values: Any = ..., keep_default_na: bool = ..., verbose: bool = ..., parse_dates: Union[bool, Sequence[Any], Dict[str, Sequence[Any]]] = ..., date_parser: Optional[Callable[..., Any]] = ..., thousands: Optional[str] = ..., comment: Optional[str] = ..., skipfooter: int = ..., convert_float: bool = ..., mangle_dupe_cols: bool = ...) -> Dict[Union[int, str], DataFrame]
tests/test_frame.py:1026: note:     def read_excel(filepath: str, sheet_name: Union[int, str] = ..., header: Union[int, Sequence[int], None] = ..., names: Optional[List[str]] = ..., index_col: Union[int, Sequence[int], None] = ..., usecols: Union[int, str, Sequence[Union[int, str, Callable[..., Any]]], None] = ..., squeeze: bool = ..., dtype: Union[str, Dict[str, Any], Union[ExtensionDtype, Union[str, dtype[generic], Type[str], Type[float], Type[int], Type[complex], Type[bool], Type[object]]]] = ..., engine: Optional[str] = ..., converters: Optional[Dict[Union[int, str], Callable[..., Any]]] = ..., true_values: Optional[Sequence[Union[str, bytes, date, datetime, timedelta, bool, int, float, complex, Timestamp, Timedelta]]] = ..., false_values: Optional[Sequence[Union[str, bytes, date, datetime, timedelta, bool, int, float, complex, Timestamp, Timedelta]]] = ..., skiprows: Union[Sequence[int], int, Callable[..., Any], None] = ..., nrows: Optional[int] = ..., na_values: Any = ..., keep_default_na: bool = ..., verbose: bool = ..., parse_dates: Union[bool, Sequence[Any], Dict[str, Sequence[Any]]] = ..., date_parser: Optional[Callable[..., Any]] = ..., thousands: Optional[str] = ..., comment: Optional[str] = ..., skipfooter: int = ..., convert_float: bool = ..., mangle_dupe_cols: bool = ..., **kwargs: Any) -> DataFrame
Found 1 error in 1 file (checked 244 source files)

===========================================
Step: 'Run Mypy Against Source Code' failed!
===========================================
```

**To Reproduce**
Call `pandas.read_excel` with `sheet_name` set to `None`.

```python
sheets: Dict[str, DataFrame] = pandas.read_excel(
    "foo", sheet_name=None
)
```

**Please complete the following information:**
 - OS: Kali / WSL2
 - OS Version 5.10.16.3-microsoft-standard-WSL2
 - 3.10.2; 3.9.9
 - pyright 1.1.256; mypy 0.961 (compiled: yes)
 - version of installed `pandas-stubs` 1.4.3.220704

--- a/pandas-stubs/io/excel/_base.pyi
+++ b/pandas-stubs/io/excel/_base.pyi
@@ -21,7 +21,7 @@ from pandas._typing import (
 @overload
 def read_excel(
     filepath: str,
-    sheet_name: List[Union[int, str]],
+    sheet_name: Union[List[Union[int, str]], None],
     header: Optional[Union[int, Sequence[int]]] = ...,
     names: Optional[List[str]] = ...,
     index_col: Optional[Union[int, Sequence[int]]] = ...,

--- a/pandas-stubs/io/excel/_base.pyi
+++ b/pandas-stubs/io/excel/_base.pyi
@@ -21,7 +21,7 @@ from pandas._typing import (
 @overload
 def read_excel(
     filepath: str,
-    sheet_name: Union[List[Union[int, str]], None],
+    sheet_name: Optional[List[Union[int, str]]],
     header: Optional[Union[int, Sequence[int]]] = ...,
     names: Optional[List[str]] = ...,
     index_col: Optional[Union[int, Sequence[int]]] = ...,

--- a/tests/test_frame.py
+++ b/tests/test_frame.py
@@ -1024,13 +1024,13 @@ def test_read_excel() -> None:
         df14: Dict[Union[int, str], pd.DataFrame] = pd.read_excel(
             "foo", sheet_name=["sheet"]
         )
-        df15: Dict[Union[int, str], pd.DataFrame] = pd.read_excel(
-            "foo", sheet_name=[0]
-        )
+        df15: Dict[Union[int, str], pd.DataFrame] = pd.read_excel("foo", sheet_name=[0])
         df16: Dict[Union[int, str], pd.DataFrame] = pd.read_excel(
             "foo", sheet_name=[0, "sheet"]
         )
-        df17: Dict[Union[int, str], pd.DataFrame] = pd.read_excel("foo", sheet_name=None)
+        df17: Dict[Union[int, str], pd.DataFrame] = pd.read_excel(
+            "foo", sheet_name=None
+        )
 
 
 def test_join() -> None:

--- a/tests/test_frame.py
+++ b/tests/test_frame.py
@@ -1020,9 +1020,17 @@ def test_read_excel() -> None:
         # https://github.com/pandas-dev/pandas-stubs/pull/33
         df11: pd.DataFrame = pd.read_excel("foo")
         df12: pd.DataFrame = pd.read_excel("foo", sheet_name="sheet")
-        df13: Dict[Union[int, str], pd.DataFrame] = pd.read_excel(
+        df13: pd.DataFrame = pd.read_excel("foo", sheet_name=0)
+        df14: Dict[Union[int, str], pd.DataFrame] = pd.read_excel(
             "foo", sheet_name=["sheet"]
         )
+        df15: Dict[Union[int, str], pd.DataFrame] = pd.read_excel(
+            "foo", sheet_name=[0]
+        )
+        df16: Dict[Union[int, str], pd.DataFrame] = pd.read_excel(
+            "foo", sheet_name=[0, "sheet"]
+        )
+        df17: Dict[Union[int, str], pd.DataFrame] = pd.read_excel("foo", sheet_name=None)
 
 
 def test_join() -> None:

--- a/tests/test_frame.py
+++ b/tests/test_frame.py
@@ -1020,10 +1020,11 @@ def test_read_excel() -> None:
         # https://github.com/pandas-dev/pandas-stubs/pull/33
         df11: pd.DataFrame = pd.read_excel("foo")
         df12: pd.DataFrame = pd.read_excel("foo", sheet_name="sheet")
-        df13: pd.DataFrame = pd.read_excel("foo", sheet_name=0)
-        df14: Dict[Union[int, str], pd.DataFrame] = pd.read_excel(
+        df13: Dict[Union[int, str], pd.DataFrame] = pd.read_excel(
             "foo", sheet_name=["sheet"]
         )
+        # GH 98
+        df14: pd.DataFrame = pd.read_excel("foo", sheet_name=0)
         df15: Dict[Union[int, str], pd.DataFrame] = pd.read_excel("foo", sheet_name=[0])
         df16: Dict[Union[int, str], pd.DataFrame] = pd.read_excel(
             "foo", sheet_name=[0, "sheet"]

